tasks/libs/ciproviders/gitlab.py

```diff
@@ -396,21 +405,141 @@ def generate_gitlab_full_configuration(input_file):
+    if compare_to:
+        for value in full_configuration.values():
+            if (
+                isinstance(value, dict)
+                and "changes" in value
+                and isinstance(value["changes"], dict)
+                and "compare_to" in value["changes"]
+            ):
+                value["changes"]["compare_to"] = compare_to
+            elif isinstance(value, list):
+                for v in value:
+                    if (
+                        isinstance(v, dict)
+                        and "changes" in v
+                        and isinstance(v["changes"], dict)
+                        and "compare_to" in v["changes"]
+                    ):
+                        v["changes"]["compare_to"] = compare_to
     return yaml.safe_dump(full_configuration)
 
 
 def read_includes(yaml_file, includes):
     """
     Recursive method to read all includes from yaml files and store them in a list
     """
-    with open(yaml_file) as f:
-        current_file = yaml.safe_load(f)
+    current_file = read_content(yaml_file)
     if 'include' not in current_file:
         includes.append(current_file)
     else:
         for include in current_file['include']:
-            if include.startswith('http'):
-                continue
             read_includes(include, includes)
         del current_file['include']
         includes.append(current_file)
+
+
+def read_content(file_path):
+    """
+    Read the content of a file, either from a local file or from an http endpoint
+    """
+    content = None
+    if file_path.startswith('http'):
+        import requests
+
+        response = requests.get(file_path)
+        response.raise_for_status()
+        content = response.text
+    else:
+        with open(file_path) as f:
+            content = f.read()
+    return yaml.safe_load(content)
+
+
+def get_preset_contexts(required_tests):
+    possible_tests = ["all", "main", "release", "mq"]
+    required_tests = required_tests.casefold().split(",")
+    if set(required_tests) | set(possible_tests) != set(possible_tests):
+        raise Exit(f"Invalid test required: {required_tests} must contain only values from {possible_tests}", 1)
+    main_contexts = [
+        ("BUCKET_BRANCH", ["nightly"]),  # ["dev", "nightly", "beta", "stable", "oldnightly"]
+        ("CI_COMMIT_BRANCH", ["main"]),  # ["main", "mq-working-branch-main", "7.42.x", "any/name"]
+        ("CI_COMMIT_TAG", [""]),  # ["", "1.2.3-rc.4", "6.6.6"]
+        ("CI_PIPELINE_SOURCE", ["pipeline"]),  # ["trigger", "pipeline", "schedule"]
+        ("DEPLOY_AGENT", ["true"]),
+        ("RUN_ALL_BUILDS", ["true"]),
+        ("RUN_E2E_TESTS", ["auto"]),
+        ("RUN_KMT_TESTS", ["on"]),
+        ("RUN_UNIT_TESTS", ["on"]),
+        ("TESTING_CLEANUP", ["true"]),
+    ]
+    release_contexts = [
+        ("BUCKET_BRANCH", ["stable"]),
+        ("CI_COMMIT_BRANCH", ["7.42.x"]),
+        ("CI_COMMIT_TAG", ["3.2.1", "1.2.3-rc.4"]),
```

Comment:
I defined the contexts this way to have the possibility to test combinations. However as the API call is quite long (~15s) we might prefer to test only few critic combinations that can be written as `yml` templates for a better readability